internal/generators/id_generator.go

```diff
@@ -0,0 +1,23 @@
+package generators
+
+import (
+	"errors"
+	"go-url-shortener/internal/storage"
+)
+
+func GenerateID(db storage.Storager, size int) (string, error) {
+	if size == 0 {
+		size = 7
+	}
+
+	id := GenerateString(size)
+
+	for step := 1; step < 10; step++ {
+		if !db.Has(id) {
+			return id, nil
+		}
+		id = GenerateString(size)
+	}
```

Comment:
nit: мне кажется так лаконичней)
```suggestion
	for step := 1; step < 10; step++ {
	        id := GenerateString(size)
		if !db.Has(id) {
			return id, nil
		}
		
	}
```